src/backend/distributed/master/worker_node_manager.c

```diff
@@ -371,36 +301,24 @@ WorkerNodeList(void)
 bool
 WorkerNodeActive(const char *nodeName, uint32 nodePort)
 {
-	bool workerNodeActive = false;
-	bool handleFound = false;
 	WorkerNode *workerNode = NULL;
-	void *hashKey = NULL;
+	HASH_SEQ_STATUS status;
+	HTAB *workerNodeHash = GetWorkerNodeHash();
 
-	WorkerNode *searchedNode = (WorkerNode *) palloc0(sizeof(WorkerNode));
-	strlcpy(searchedNode->workerName, nodeName, WORKER_LENGTH);
-	searchedNode->workerPort = nodePort;
+	hash_seq_init(&status, workerNodeHash);
 
-	hashKey = (void *) searchedNode;
-	workerNode = (WorkerNode *) hash_search(WorkerNodesHash, hashKey,
```

Comment:
I'm guessing we're using a hash here for historical reasons, but after this change we never look anything up in the hash. 

Would if we use nodename/nodeport as the key for the hash, or use a list?